python/cudf/cudf/_lib/pylibcudf/strings/find.pxd

```diff
@@ -3,10 +3,8 @@
 from cudf._lib.pylibcudf.column cimport Column
 from cudf._lib.pylibcudf.libcudf.types cimport size_type
 from cudf._lib.pylibcudf.scalar cimport Scalar
+from cudf._lib.pylibcudf.strings.types cimport ColumnOrScalar
```

Comment:
I like that you went here. We do have a lot of fused types proliferating. I considered consolidating them in a common header earlier, but I got wary of doing that out of concern for creating circular import issues. Can we undo that change in this PR, then do a proper analysis in a separate PR where we see which fused types can safely be consolidated and where? I suspect we'll have to do something like have a top-level types.pxd that is just fused types based on base types (Column, Table, Scalar) and then some more specific modules that declare more specific fused types because those specific types won't be safe to put in the top-level one without creating circular cimports.